src/concrete/ml/sklearn/xgb.py

```diff
@@ -391,6 +445,19 @@ def dump_dict(self) -> Dict[str, Any]:
         metadata["use_label_encoder"] = self.use_label_encoder
         metadata["random_state"] = self.random_state
         metadata["verbosity"] = self.verbosity
+        metadata["max_bin"] = self.max_bin
+        metadata["max_leaves"] = self.max_leaves
+        metadata["max_cat_to_onehot"] = self.max_cat_to_onehot
+        metadata["grow_policy"] = self.grow_policy
+        metadata["sampling_method"] = self.sampling_method
+        metadata["early_stopping_rounds"] = self.early_stopping_rounds
+        metadata["eval_metric"] = self.eval_metric
+
+        # Callables are not serializable
+        assert not self.kwargs, "kwargs are not supported for serialization"
```

Comment:
cf comment above, we should start having valuerror + tests instead of assert with user inputs (refs https://github.com/zama-ai/concrete-ml-internal/issues/4232)

Reply:
Done